misc: warnings on MPI_STATUS_IGNORE etc with  gcc-11.2.1

I'm not sure if this is a compiler issue - or code isse [occurs with mpich - but not openmpi]

This is a 32bit build with `gcc (GCC) 11.2.1 20210728 (Red Hat 11.2.1-1)` on Fedora 35

```
[glci@pj03 petsc]$ make V=1 PETSC_ARCH=arch-ci-linux-gcc-complex-opt-32bit arch-ci-linux-gcc-complex-opt-32bit/obj/sys/utils/mpits.o
/home/glci/petsc-hash-pkgs/494b06380/bin/mpicc -c -m32 -fPIC -Wall -Wwrite-strings -Wno-strict-aliasing -Wno-unknown-pragmas -g -O    -I/home/glci/petsc/include -I/home/glci/petsc/arch-ci-linux-gcc-complex-opt-32bit/include -I/home/glci/petsc-hash-pkgs/494b06380/include    -MMD -MP /home/glci/petsc/src/sys/utils/mpits.c -o arch-ci-linux-gcc-complex-opt-32bit/obj/sys/utils/mpits.o
/home/glci/petsc/src/sys/utils/mpits.c: In function ‘PetscCommBuildTwoSidedFReq_Ibarrier’:
/home/glci/petsc/src/sys/utils/mpits.c:425:14: warning: ‘MPI_Testall’ accessing 20 bytes in a region of size 0 [-Wstringop-overflow=]
  425 |       ierr = MPI_Testall(nsends,sendreqs,&sent,MPI_STATUSES_IGNORE);CHKERRMPI(ierr);
      |              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/glci/petsc/src/sys/utils/mpits.c:425:14: note: referencing argument 4 of type ‘MPI_Status *’
In file included from /home/glci/petsc/include/petscsys.h:162,
                 from /home/glci/petsc/src/sys/utils/mpits.c:1:
/home/glci/petsc-hash-pkgs/494b06380/include/mpi.h:947:5: note: in a call to function ‘MPI_Testall’
  947 | int MPI_Testall(int count, MPI_Request array_of_requests[], int *flag,
      |     ^~~~~~~~~~~
/home/glci/petsc/src/sys/utils/mpits.c: In function ‘PetscCommBuildTwoSided_Ibarrier’:
/home/glci/petsc/src/sys/utils/mpits.c:125:14: warning: ‘MPI_Testall’ accessing 20 bytes in a region of size 0 [-Wstringop-overflow=]
  125 |       ierr = MPI_Testall(nsends,sendreqs,&sent,MPI_STATUSES_IGNORE);CHKERRMPI(ierr);
      |              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/glci/petsc/src/sys/utils/mpits.c:125:14: note: referencing argument 4 of type ‘MPI_Status *’
In file included from /home/glci/petsc/include/petscsys.h:162,
                 from /home/glci/petsc/src/sys/utils/mpits.c:1:
/home/glci/petsc-hash-pkgs/494b06380/include/mpi.h:947:5: note: in a call to function ‘MPI_Testall’
  947 | int MPI_Testall(int count, MPI_Request array_of_requests[], int *flag,
      |     ^~~~~~~~~~~
In file included from /home/glci/petsc/include/petscsys.h:1674,
                 from /home/glci/petsc/src/sys/utils/mpits.c:1:
/home/glci/petsc/src/sys/utils/mpits.c: In function ‘PetscCommBuildTwoSidedF’:
/home/glci/petsc/include/petsclog.h:551:81: warning: ‘MPI_Waitall’ accessing 20 bytes in a region of size 0 [-Wstringop-overflow=]
  551 |   ((petsc_wait_all_ct++,petsc_sum_of_waits_ct += (PetscLogDouble) (count),0) || MPI_Waitall((count),(array_of_requests),(array_of_statuses)))
      |                                                                                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/glci/petsc/src/sys/utils/mpits.c:494:10: note: in expansion of macro ‘MPI_Waitall’
  494 |   ierr = MPI_Waitall(nto*ntags,toreqs,MPI_STATUSES_IGNORE);CHKERRMPI(ierr);
      |          ^~~~~~~~~~~
/home/glci/petsc/include/petsclog.h:551:81: note: referencing argument 3 of type ‘MPI_Status *’
  551 |   ((petsc_wait_all_ct++,petsc_sum_of_waits_ct += (PetscLogDouble) (count),0) || MPI_Waitall((count),(array_of_requests),(array_of_statuses)))
      |                                                                                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/glci/petsc/src/sys/utils/mpits.c:494:10: note: in expansion of macro ‘MPI_Waitall’
  494 |   ierr = MPI_Waitall(nto*ntags,toreqs,MPI_STATUSES_IGNORE);CHKERRMPI(ierr);
      |          ^~~~~~~~~~~
In file included from /home/glci/petsc/include/petscsys.h:162,
                 from /home/glci/petsc/src/sys/utils/mpits.c:1:
/home/glci/petsc-hash-pkgs/494b06380/include/mpi.h:946:5: note: in a call to function ‘MPI_Waitall’
  946 | int MPI_Waitall(int count, MPI_Request array_of_requests[], MPI_Status array_of_statuses[]) MPICH_API_PUBLIC;
      |     ^~~~~~~~~~~
In file included from /home/glci/petsc/include/petscsys.h:1674,
                 from /home/glci/petsc/src/sys/utils/mpits.c:1:
/home/glci/petsc/include/petsclog.h:551:81: warning: ‘MPI_Waitall’ accessing 20 bytes in a region of size 0 [-Wstringop-overflow=]
  551 |   ((petsc_wait_all_ct++,petsc_sum_of_waits_ct += (PetscLogDouble) (count),0) || MPI_Waitall((count),(array_of_requests),(array_of_statuses)))
      |                                                                                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/glci/petsc/src/sys/utils/mpits.c:495:10: note: in expansion of macro ‘MPI_Waitall’
  495 |   ierr = MPI_Waitall(*nfrom*ntags,fromreqs,MPI_STATUSES_IGNORE);CHKERRMPI(ierr);
      |          ^~~~~~~~~~~
/home/glci/petsc/include/petsclog.h:551:81: note: referencing argument 3 of type ‘MPI_Status *’
  551 |   ((petsc_wait_all_ct++,petsc_sum_of_waits_ct += (PetscLogDouble) (count),0) || MPI_Waitall((count),(array_of_requests),(array_of_statuses)))
      |                                                                                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/glci/petsc/src/sys/utils/mpits.c:495:10: note: in expansion of macro ‘MPI_Waitall’
  495 |   ierr = MPI_Waitall(*nfrom*ntags,fromreqs,MPI_STATUSES_IGNORE);CHKERRMPI(ierr);
      |          ^~~~~~~~~~~
In file included from /home/glci/petsc/include/petscsys.h:162,
                 from /home/glci/petsc/src/sys/utils/mpits.c:1:
/home/glci/petsc-hash-pkgs/494b06380/include/mpi.h:946:5: note: in a call to function ‘MPI_Waitall’
  946 | int MPI_Waitall(int count, MPI_Request array_of_requests[], MPI_Status array_of_statuses[]) MPICH_API_PUBLIC;
      |     ^~~~~~~~~~~
```

I see mpi.h has:
```
#define MPI_STATUSES_IGNORE (MPI_Status *)1
```
Changing this '0' gets rid of the warning (openmpi uses this) [same issue with 'MPI_STATUS_IGNORE']
```
#define MPI_STATUSES_IGNORE (MPI_Status *)0
```

```
[glci@pj03 petsc]$ make V=1 PETSC_ARCH=arch-ci-linux-gcc-complex-opt-32bit arch-ci-linux-gcc-complex-opt-32bit/obj/sys/utils/mpits.o
/home/glci/petsc-hash-pkgs/494b06380/bin/mpicc -c -m32 -fPIC -Wall -Wwrite-strings -Wno-strict-aliasing -Wno-unknown-pragmas -g -O    -I/home/glci/petsc/include -I/home/glci/petsc/arch-ci-linux-gcc-complex-opt-32bit/include -I/home/glci/petsc-hash-pkgs/494b06380/include    -MMD -MP /home/glci/petsc/src/sys/utils/mpits.c -o arch-ci-linux-gcc-complex-opt-32bit/obj/sys/utils/mpits.o
[glci@pj03 petsc]$ /home/glci/petsc-hash-pkgs/494b06380/bin/mpichversion 
MPICH Version:          3.4.2
MPICH Release date:     Wed May 26 15:51:40 CDT 2021
MPICH Device:           ch3:nemesis
MPICH configure:        --prefix=/home/glci/petsc-hash-pkgs/494b06380 MAKE=/usr/bin/gmake --libdir=/home/glci/petsc-hash-pkgs/494b06380/lib CC=gcc CFLAGS=-m32 -fPIC -g -O AR=/usr/bin/ar ARFLAGS=cr CXX=g++ CXXFLAGS=-m32 -fPIC -g -O -fPIC -std=gnu++17 FFLAGS=-m32 -fPIC -ffree-line-length-0 -g -O -fallow-argument-mismatch FC=gfortran F77=gfortran FCFLAGS=-m32 -fPIC -ffree-line-length-0 -g -O -fallow-argument-mismatch --enable-shared --with-pm=hydra --disable-java --with-device=ch3:nemesis --enable-g=meminit
MPICH CC:       gcc -m32 -fPIC -g -O   -O2
MPICH CXX:      g++ -m32 -fPIC -g -O -fPIC -std=gnu++17  -O2
MPICH F77:      gfortran -m32 -fPIC -ffree-line-length-0 -g -O -fallow-argument-mismatch  -O2
MPICH FC:       gfortran -m32 -fPIC -ffree-line-length-0 -g -O -fallow-argument-mismatch  -O2
MPICH Custom Information: 
[glci@pj03 petsc]$ 
```

--- a/maint/local_python/binding_common.py
+++ b/maint/local_python/binding_common.py
@@ -139,6 +139,10 @@ def get_C_param(param, func, mapping):
         else:
             # MPI_Init, MPI_Init_thread -> char ***argv
             want_star = 3
+    elif kind == "STATUS" and param['length'] is not None:
+        # MPI_{Wait,Test}{all,some}
+        # gcc-11 warns when we pass MPI_STATUSES_IGNORE to MPI_Status statues[]
+        want_star = 1
     elif not want_star:
         if is_pointer_type(param):
             if kind == "STRING_ARRAY":

